packages/atoms/connect/ConnectButton.tsx

```diff
@@ -0,0 +1,55 @@
+import { useState } from "react";
+
+import { classNames } from "@calcom/lib";
+
+import { Button } from "../src/components/ui/button";
+import type { AtomsGlobalConfigProps } from "../types";
+import type { ConnectButtonProps } from "../types";
+
+// we accept two props: buttonText & onButtonClick
+// buttonText is for displaying buttonText
+// onButtonClick is a a method to handle onClick for button
+// it accepts another object which contains a callback in case of any error
+
+export function ConnectButton({
+  buttonText,
+  onButtonClick,
+  stylesClassname,
+  icon,
+}: ConnectButtonProps & AtomsGlobalConfigProps) {
```

Comment:
here I think we should consider extending the props of a basic HTML button like in shadcn/ui component, 

this is kind of related to the Liskov substitution principle, meaning that since we have providing a wrapper around the basic HTML button, we should provide a similar interface 

this is how it's done in shadcn/ui ` React.forwardRef<HTMLButtonElement, ButtonProps>`

Reply:
And then we can pass in additional props inside of the button like this `<Button ...otherProps>` right?

Reply:
yes exactly, like it is done inside shadcn/ui button component